### ISSUE TITLE
Unable to add a duplicate FQDN in provision when Overseer and FL servers specify the same FQDN.

### ISSUE THREAD

**Issue by YuanTingHsieh** [NVIDIA/NVFlare#777]
### Discussed in https://github.com/NVIDIA/NVFlare/discussions/710

<div type='discussions-op-text'>

<sup>Originally posted by **asus-ocis** July  6, 2022</sup>
Hi all,

I want to deploy nvflare 2.1 on k8s. Overseer, server1 and server2 will run on different pods,
We have a HAproxy on k8s, so we need to set the same FQDN.

my project requirements:
```
Overseer : 
  fqdn : myk8s.com
  port: 7001
server1:
   fqdn: myk8s.com
   learn_port:70032
   admin_port:70033
server2:
   fqdn: myk8s.com
   learn_port:70055
   admin_port:70052   

```
But I got the following error after executing provison.
`ValueError: Unable to add a duplicate name myk8s.com into this project.`

Check spec.py
https://github.com/NVIDIA/NVFlare/blob/a23afd7759b36f3885c9de68b461c5ea8358d085/nvflare/lighter/spec.py#L72
The name field in the participants needs to be unique.

Is it possible to add FQDN field for Overseer and FL Server in project.yml in future versions? Or have any other suggestion?

</div>

**Comment by parkeraddison** [comment]
Also ran into this. For now one workaround is to set the participant name/fqdn to something fake like "server.hostname" and "overseer.hostname" then add the real IP mapping to the /etc/hosts file on each participant.

**Comment by YuanTingHsieh** [comment]
> I am running into the same issue, as I want to setup Overseer and Server on the same FQDN. Is there any reason, why this is not enabled (or considered bad practice)?

@IsaacYangSLA and @chesterxgchen can you help answer this question? thanks!

**Comment by parkeraddison** [comment]
> In case of there are two FL servers, one dead, if the Overseer is on the same host of FL Server. The Overseer could crash at the same time if the host hardware is no longer working.

I agree it's a bad practice to put the Server and Overseer on the same physical host -- but same FQDN doesn't necessary imply same host. E.g. if a load balancer is used in a cloud or k8s deployment, then the same FQDN can forward to different hosts depending on the port.

FWIW if we provision using the Dashboard then the same FQD can be used (just might help to rename for the files when you download them to avoid confusion).

Can't do this in a project.yaml because the service's name is its FQDN... could be worth separating the two fields like was proposed in the original post.
